### DISCUSSION TITLE
How are you backing up your SD card?

### DISCUSSION THREAD

**Discussion by Jkh62** [jgyates/genmon#1057] [General]
Pulling it and making a image?
Backing up files by SSH over network? 
USB slot to card on pi??

**Comment by skipfire** [comment]
I download the genmon configs to my PC in a cloud sync'd folder after I make changes, other than that I don't backup as I will just restore from scratch if necessary since it doesn't take too long and is well documented.

**Comment by jgyates** [comment]
same, I don't back up the SD card as I can get a new pi image on a SD card in 10 min, then just install genmon with the install script, then restore the settings. 

I just backup the genmon config files that you can download on the about page. You can restore from a back up per these instructions: https://github.com/jgyates/genmon/wiki/Appendix-M-Tips-and-Tricks-useful-for-Genmon#restoring-settings-from-a-backup

**Comment by Jkh62** [comment]
I did the same.
I made a card image from my card reader before the last update and after it was loaded and hooked up. 

Last update is nice with the added L1 /L2 BTW.

**Comment by skipfire** [comment]
Another suggestion is to track the commands you use, and then create scripts for them.  genmonmaint.sh does have switches so you can potentially automate the full install.  I can put the OS on a fresh micro-SD card, then run the following commands and the only this missing is restoring my config files.  The top 6 lines of this script install Log2Ram and is optional, the last 3 lines do genmon.
```
echo "deb [signed-by=/usr/share/keyrings/azlux-archive-keyring.gpg] http://packages.azlux.fr/debian/ bookworm main" | sudo tee /etc/apt/sources.list.d/azlux.list
sudo wget -O /usr/share/keyrings/azlux-archive-keyring.gpg  https://azlux.fr/repo.gpg
sudo apt update -y
sudo apt install log2ram
sudo sed -i "s/SIZE=40M/SIZE=250M/g" /etc/log2ram.conf
sudo sed -i "s/LOG_DISK_SIZE=100M/LOG_DISK_SIZE=300M/g" /etc/log2ram.conf

git clone https://github.com/jgyates/genmon.git
cd ~/genmon
bash ./genmonmaint.sh -i -n
```
The -n switch covers most standard installs on a Pi with choosing the UART serial pins and yes to the prompts.  If you answer no to any of the prompts or pick TCP or USB connection instead of serial then you will not want to use the -n switch.

**Comment by Jkh62** [comment]
I added the - n for no prompt and everything ran / installed perfectly.
You are / were very through and exact in your instructions. 

The raspberry pi is new to me and I have no clue to longevity. 
I am actually considering getting a spare for my current HAT or another one so I have one ready to swap out if it goes down. Just added UPS battery backups for my modem and router so no loss sending email alerts when the power drops. 

I also have Log2ram installed and running. Seems like a smart thing to do. 
Thus far no com errors, time outs or dropped packets. 

Really impressed with Genmon.
